routers/api/v1/api.go

```diff
@@ -1173,7 +1209,7 @@ func Routes(ctx gocontext.Context) *web.Route {
 				m.Post("/{username}/{reponame}", admin.AdoptRepository)
 				m.Delete("/{username}/{reponame}", admin.DeleteUnadoptedRepository)
 			})
-		}, reqToken(), reqSiteAdmin())
+		}, reqSiteAdminOrToken())
```

Comment:
I think original code means both are required.

Reply:
In [tests/integration/api_httpsig_test.go](https://github.com/go-gitea/gitea/pull/20908/files/525c11d282141fec4c1cf9cbdfdd6804b0f0f860#diff-94c5559f6a55e18d4bc1c6b97b79a2a3433280b72bdfceb2b5a43a1faec8bac1):

	req = NewRequest(t, "GET", "/api/v1/admin/users")

Then I don't really understand what do you mean by the above test ['should not use token but signature to auth'](https://github.com/go-gitea/gitea/pull/20908#discussion_r1014622287). If token is required, then I think we should update the above test and use token then?